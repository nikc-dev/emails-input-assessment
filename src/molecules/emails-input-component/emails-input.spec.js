import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EmailsInput from './index';



it('should render EmailsInput correctly', () => {
    document.body.innerHTML =
    '<div id="emailsInput__emai-input">' +
    '  <span  />' +
    '</div>';
  const component = shallow(
    <EmailsInput updateEmailsList={jest.fn()} removeEmail={jest.fn()} emailsToRender={[{emailId: 'test@test.com'}]} />,
  );
  component.find('#emailsInput').simulate('click');
  expect(toJson(component)).toMatchSnapshot();
});

it('should accpet valid Email Input correctly', () => {
  const component = shallow(
    <EmailsInput updateEmailsList={jest.fn()} removeEmail={jest.fn()} emailsToRender={[{emailId: 'test@test.com'}]} />,
  );
  component.find('input').simulate('keyDown', { target: { value: 'test@test.com' }, key: 'Enter' });
  expect(toJson(component)).toMatchSnapshot();
});

it('should split pasted emails that is comma seperated', () => {
  const component = shallow(
    <EmailsInput updateEmailsList={jest.fn()} removeEmail={jest.fn()} updatePastedEmails={jest.fn()} /> 
  );
    component.find('input').simulate("paste", { clipboardData: { getData: jest.fn().mockReturnValue("test@test.com,test2@test.com")}})
    expect(toJson(component)).toMatchSnapshot();
})

it('should accpet Invalid Email Input correctly', () => {
    const component = shallow(
      <EmailsInput updateEmailsList={jest.fn()} removeEmail={jest.fn()} emailsToRender={[{emailId: 'test@test.com'}]} />,
    );
    component.find('input').simulate('keyDown', { target: { value: 'test@test' }, key: 'Tab' });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call onChange function on value change', () => {
    const component = shallow(
      <EmailsInput updateEmailsList={jest.fn()} removeEmail={jest.fn()} updatePastedEmails={jest.fn()} /> 
    );
      component.find('input').simulate("change", { target: {value: 'abc'}});
      expect(toJson(component)).toMatchSnapshot();
  })

  it('should accpet Email Input correctly on pressing comma', () => {
    const component = shallow(
      <EmailsInput updateEmailsList={jest.fn()} removeEmail={jest.fn()} emailsToRender={[{emailId: 'test@test.com'}]} />,
    );
    component.find('input').simulate('keyDown', { target: { value: 'test@test' }, key: ',' });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should skip Email Input entry correctly on pressing Enter with empty value', () => {
    const component = shallow(
      <EmailsInput updateEmailsList={jest.fn()} removeEmail={jest.fn()} emailsToRender={[{emailId: 'test@test.com'}]} />,
    );
    component.find('input').simulate('keyDown', { target: { value: '' }, key: 'Enter' });
    expect(toJson(component)).toMatchSnapshot();
  });
