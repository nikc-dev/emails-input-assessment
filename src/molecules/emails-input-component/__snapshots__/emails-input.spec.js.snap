// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accpet Email Input correctly on pressing comma 1`] = `
<div
  className="emailsInput"
  id="emailsInput"
  onClick={[Function]}
>
  <EmailChips
    email={
      Object {
        "emailId": "test@test.com",
      }
    }
    key="0"
    removeEmail={[MockFunction]}
  />
  <input
    autoComplete="off"
    className="emailsInput__input"
    id="emailsInput__emai-input"
    name="email"
    onChange={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholder="Add people..."
    type="email"
    value=""
  />
</div>
`;

exports[`should accpet Invalid Email Input correctly 1`] = `
<div
  className="emailsInput"
  id="emailsInput"
  onClick={[Function]}
>
  <EmailChips
    email={
      Object {
        "emailId": "test@test.com",
      }
    }
    key="0"
    removeEmail={[MockFunction]}
  />
  <input
    autoComplete="off"
    className="emailsInput__input"
    id="emailsInput__emai-input"
    name="email"
    onChange={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholder="Add people..."
    type="email"
    value=""
  />
</div>
`;

exports[`should accpet valid Email Input correctly 1`] = `
<div
  className="emailsInput"
  id="emailsInput"
  onClick={[Function]}
>
  <EmailChips
    email={
      Object {
        "emailId": "test@test.com",
      }
    }
    key="0"
    removeEmail={[MockFunction]}
  />
  <input
    autoComplete="off"
    className="emailsInput__input"
    id="emailsInput__emai-input"
    name="email"
    onChange={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholder="Add people..."
    type="email"
    value=""
  />
</div>
`;

exports[`should call onChange function on value change 1`] = `
<div
  className="emailsInput"
  id="emailsInput"
  onClick={[Function]}
>
  <input
    autoComplete="off"
    className="emailsInput__input"
    id="emailsInput__emai-input"
    name="email"
    onChange={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholder="Add people..."
    type="email"
    value="abc"
  />
</div>
`;

exports[`should render EmailsInput correctly 1`] = `
<div
  className="emailsInput"
  id="emailsInput"
  onClick={[Function]}
>
  <EmailChips
    email={
      Object {
        "emailId": "test@test.com",
      }
    }
    key="0"
    removeEmail={[MockFunction]}
  />
  <input
    autoComplete="off"
    className="emailsInput__input"
    id="emailsInput__emai-input"
    name="email"
    onChange={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholder="Add people..."
    type="email"
    value=""
  />
</div>
`;

exports[`should skip Email Input entry correctly on pressing Enter with empty value 1`] = `
<div
  className="emailsInput"
  id="emailsInput"
  onClick={[Function]}
>
  <EmailChips
    email={
      Object {
        "emailId": "test@test.com",
      }
    }
    key="0"
    removeEmail={[MockFunction]}
  />
  <input
    autoComplete="off"
    className="emailsInput__input"
    id="emailsInput__emai-input"
    name="email"
    onChange={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholder="Add people..."
    type="email"
    value=""
  />
</div>
`;

exports[`should split pasted emails that is comma seperated 1`] = `
<div
  className="emailsInput"
  id="emailsInput"
  onClick={[Function]}
>
  <input
    autoComplete="off"
    className="emailsInput__input"
    id="emailsInput__emai-input"
    name="email"
    onChange={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholder="Add people..."
    type="email"
    value=""
  />
</div>
`;
